docs/servo_power_monitor/src/index.ts

```diff
@@ -122,26 +147,33 @@ function setupStartUSBButton() {
     }
     if (!device) {
       device = null;
-      requestUSBButton.disabled = false;
       return;
     }
-
+    
     try {
       await device.open();
+      requestUSBButton.disabled = true;
       await device.selectConfiguration(1);
       await device.claimInterface(usb_interface);
       kickWriteLoop(async (s) => {
         let data = new TextEncoder().encode(s);
         await device.transferOut(ep, data);
       })
       readLoop(async () => {
-        let result = await device.transferIn(ep, 64);
-        if (result.status === 'stall') {
-          await device.clearHalt('in', ep);
-          throw result;
+        try {
+          let result = await device.transferIn(ep, 64);
+          if (result.status === 'stall') {
+            await device.clearHalt('in', ep);
+            throw result;
+          }
+          const result_array = new Int8Array(result.data.buffer);
+          return utf8decoder.decode(result_array);
+        } catch (e) {
+          if (!halt) {
```

Comment:
Adding a comment to explain why we are checking the halt flag here and why it is safe to ignore the error when halt is true is a good practice